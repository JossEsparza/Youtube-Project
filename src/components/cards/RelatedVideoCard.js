import React from 'react';
import './RelatedVideoCard.scss';

const RelatedVideoCard = ({ snippet }) => {
  const { title, channelTitle, thumbnails } = snippet;
  const { medium } = thumbnails;
  return (
    <div className="related_video_container">
      <img src={medium.url} alt={title} />
      <div className="related_video_content">
        <h4>{title}</h4>
        <span>Channel: {channelTitle}</span>
      </div>
    </div>
  );
};

export default RelatedVideoCard;
