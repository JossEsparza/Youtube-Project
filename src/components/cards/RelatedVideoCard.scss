.related_video_container {
  display: flex;
  flex-direction: row;
  justify-content: left;
}

.related_video_content {
  display: flex;
  flex-direction: column;
}

h4 {
  width: 230px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

img {
  width: 250px;
  margin-bottom: 7px;
}
