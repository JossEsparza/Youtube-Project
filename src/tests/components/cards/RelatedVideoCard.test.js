import { shallow } from 'enzyme';
import RelatedVideoCard from '../../../components/cards/RelatedVideoCard';

describe('Tests the RelatedVideoCard', () => {
  test('should be displayed correctly', () => {
    const id = {
      videoId: 'ToI6UOZZSzo',
    };

    const snippet_info = {
      title: 'Djokovic Vs Monfils; Alcaraz, Murray & Shapovalov In Action',
      description: 'Highlights of a blockbuster day in Madrid',
      thumbnails: {
        medium: {
          url: 'https://i.ytimg.com/vi/ToI6UOZZSzo/mqdefault.jpg',
        },
      },
    };
    const wrapper = shallow(<RelatedVideoCard id={id} snippet={snippet_info} />);
    expect(wrapper).toMatchSnapshot();
  });
});
