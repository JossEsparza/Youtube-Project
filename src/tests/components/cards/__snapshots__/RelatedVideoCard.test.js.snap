// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the RelatedVideoCard should be displayed correctly 1`] = `ShallowWrapper {}`;
